src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -118,6 +117,15 @@ public void checkout(
             @CheckForNull SCMRevisionState baseline)
             throws IOException, InterruptedException {
         gitSCM.checkout(build, launcher, workspace, listener, changelogFile, baseline);
+
+        BitbucketServerConfiguration server = getServer();
+        bitbucketClientFactoryProvider.getClient(server, CredentialUtils.getCredentials(server.getCredentialsId()))
+                .getBuildStatusClient(getLatestRevision(build))
+                .post(new BitbucketBuildStatus((Build) build));
```

Comment:
Is this an unsafe cast? I assume all calls to checkout will have runs that are instances of `FreeStyleBuild`

Reply:
Not safe, the pipeline builds will not be `FreeStyleBuild` and we must support pipeline builds.

Reply:
Can we assume at least `AbstractBuild` (rather than run)?